src/coreclr/jit/compiler.h

```diff
@@ -3455,7 +3455,7 @@ class Compiler
 #endif
 #endif // FEATURE_HW_INTRINSICS
 
-    GenTree* gtNewMemoryBarrier(bool loadOnly = false);
+    GenTree* gtNewMemoryBarrier(bool loadOnly = false, bool storeOnly = false);
```

Comment:
It could be better to use an enum here and not rely on asserts that both flags can be set. Can this take `BarrierKind` as an argument?